### PR TITLE
feat(treeview): custom icons with img

### PR DESCRIPTION
**Pull request by jsomsanith**
**What is the problem this PR is trying to solve?**
Set an icon from a plain link in a tree view (item).

**What is the chosen solution to this problem?**
Support src in Icon and wire options in treeviewitem.

capture d ecran de 2018-02-02 12-01-37

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js [hunk 1]
```diff
@@ -23,6 +23,28 @@ export function getItemIcon(iconName = 'talend-folder', toggled) {
...
+ */
+function TreeViewIcon({ icon, toggled }) {
+	const isObject = typeof icon === 'object';
+	const iconConfig = isObject ?
+		{ ...icon, className: classNames(css['tc-treeview-img'], icon.className) } :
+		{ name: getItemIcon(icon, toggled), className: css['tc-treeview-folder'] };
+
+	return (
+		<Icon {...iconConfig} />
+	);
```
Comment (talend-glorieux) [inline packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js]: Please use a regular `if/else` statement that returns an `Icon` with the proper props.

File: packages/components/src/Icon/Icon.component.js [hunk 2]
```diff
@@ -34,14 +34,23 @@ export const TRANSFORMS = Object.keys(FA_TRANSFORMS);
...
 		title: title || null,
 	};
+	if (src) {
+		const classNames = classnames(
+			theme['tc-icon'],
+			'tc-icon',
+			className
+		);
+
+		return <img className={classNames} src={src} alt={title} {...accessibility} />;
```
Comment (talend-glorieux) [inline packages/components/src/Icon/Icon.component.js]: `alt` should have a default value is `title` isn't set.

File: packages/components/src/Icon/Icon.component.js [hunk 3]
```diff
@@ -34,14 +34,19 @@ export const TRANSFORMS = Object.keys(FA_TRANSFORMS);
...
+function Icon({ className, name, title, transform, onClick, src }) {
 	const accessibility = {
 		focusable: 'false', // IE11
 		'aria-hidden': 'true',
 		title: title || null,
 	};
+	if (src) {
+		const classNames = classnames(theme['tc-icon'], 'tc-icon', className);
+
+		return <img className={classNames} src={src} alt={title || src} {...accessibility} />;
```
Comment (frassinier) [inline packages/components/src/Icon/Icon.component.js]: No, remove alt content here please

File: packages/components/src/Icon/__snapshots__/Icon.test.js.snap [hunk 4]
```diff
@@ -18,6 +18,17 @@ exports[`Icon should render from custom font 1`] = `
 />
 `;
 
+exports[`Icon should render from src 1`] = `
+<img
+  alt="/foo/bar.png"
```
Comment (frassinier) [inline packages/components/src/Icon/__snapshots__/Icon.test.js.snap]: no, should be empty

File: packages/components/src/Icon/__snapshots__/Icon.test.js.snap [hunk 5]
```diff
@@ -18,6 +18,17 @@ exports[`Icon should render from custom font 1`] = `
...
+  aria-hidden="true"
+  className="theme-tc-icon tc-icon"
+  focusable="false"
```
Comment (frassinier) [inline packages/components/src/Icon/__snapshots__/Icon.test.js.snap]: no sense

File: packages/components/src/Icon/__snapshots__/Icon.test.js.snap [hunk 6]
```diff
@@ -18,6 +18,17 @@ exports[`Icon should render from custom font 1`] = `
...
+  src="/foo/bar.png"
+  title={null}
```
Comment (frassinier) [inline packages/components/src/Icon/__snapshots__/Icon.test.js.snap]: no sense

**Pull request merged**